src/app/components/elements/inputs/PasswordInput.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import {
+  FormFeedback,
+  FormGroup,
+  Input,
+  Label,
+  UncontrolledTooltip,
+} from "reactstrap";
+import { PasswordFeedback, ValidationUser } from "../../../../IsaacAppTypes";
+import { Immutable } from "immer";
+import { passwordDebounce, validatePassword } from "../../../services";
+
+interface PasswordInputProps {
+  fieldType: "password" | "confirmPassword";
+  userToUpdate: Immutable<ValidationUser>;
+  setUserToUpdate: (user: Immutable<ValidationUser>) => void;
+  setUnverifiedPassword: (password: string) => void;
+  submissionAttempted: boolean;
+  unverifiedPassword: string | undefined;
+  defaultPassword?: string;
+}
+
+export const PasswordInput = ({
+  fieldType,
+  userToUpdate,
+  setUserToUpdate,
+  unverifiedPassword,
+  setUnverifiedPassword,
+  submissionAttempted,
+  defaultPassword,
+}: PasswordInputProps) => {
+  const [passwordFeedback, setPasswordFeedback] =
+    useState<PasswordFeedback | null>(null);
+
+  const passwordIsValid =
+    userToUpdate.password == unverifiedPassword &&
+    validatePassword(userToUpdate.password || "");
+
+  const newPassword = (
+    <FormGroup>
+      <Label htmlFor="password-input">Password</Label>
+      <span id={`password-help-tooltip`} className="icon-help ml-1" />
+      <UncontrolledTooltip target={`password-help-tooltip`} placement="bottom">
+        {
+          "Passwords must be at least 12 characters, containing at least one number, one lowercase letter, one uppercase letter, and one special character."
```

Comment:
We actually only allow quite a narrow range of "special" characters. It might be worth presenting them here, otherwise people will be typing e.g. æ and getting it rejected

Reply:
I did think about this, but given that the PW requirement changes were for security I wasn't sure that we ought to spell out exactly which characters we expected? I find the majority of websites don't explicitly state these in their password requirements too.